docs/src/2.5.AddingInteractivity.mdx

```diff
@@ -0,0 +1,116 @@
+    orientation: { x: 0, y: 0, z: 0, w: 1 },
+    position: { x: 0, y: 0, z: 0 },
+  },
+  scale: { x: 1, y: 1, z: 1 },
+  colors: [],
+  points: [],
+};
+```
+
+Let's also enable manual camera movement, so we can see the duck better.
+
+```js
+<Worldview defaultCameraState={{ ...DEFAULT_CAMERA_STATE, distance: 120 }}>
+```
+
+Next, let's add a `clickedObjectIds` state to keep track of the clicked spheres. We'll also use this `clickedObjectIds` state to generate cube obstacles.
+
+```js
+const [clickedObjectIds, setClickedObjectIds] = useState(new Set());
+
+const obstacleMarkers = Array.from(clickedObjectIds).map((clickedObjectId, index) => {
+  const pointIdx = clickedObjectId - marker.id;
+  const position = marker.points[pointIdx];
+  return {
+    // keep the id unique by adding sphere's id range as offset: marker.id + steps
+    id: marker.id + steps + index,
+    // remember the original clickedObjectId so when the obstacle is clicked, we can
+    // remove the obstacle quickly by updating clickedObjectIds
+    clickedObjectId,
+    pose: {
+      orientation: { x: 0, y: 0, z: 0, w: 1 },
+      position: position,
+    },
+    color: { r: 1, g: 0, b: 0, a: 1 }, // red
+    scale: { x: 4, y: 4, z: 4 }, // scale up a little so it's bigger than the spheres
+  };
+});
+```
+
+Now, let's use the clicked sphere id to update the `clickedObjectIds` state, which will be consumed by the cube obstacles. Clicking each obstacle should remove it from the scene.
+
+```js
+ <Spheres
+    onClick={(ev, { objectId, object }) => {
+        // click on the object to add an obstacle
+        clickedObjectIds.add(objectId);
+        setClickedObjectIds(new Set(clickedObjectIds));
+    }}>
+    {[marker]}
+    </Spheres>
+    <Cubes
+        onClick={(ev, { objectId, object }) => {
+            // the `object` contains information of the clicked cube, which has `clickedObjectId`
+            clickedObjectIds.delete(object.clickedObjectId);
+            setClickedObjectIds(new Set(clickedObjectIds));
+        }}>
+        {obstacleMarkers}
+    </Cubes>
+```
+
+Lastly, setting an `shouldStopDuck` variable will allow us to tell the duck when to stop and when to move forward.
+
+```js
+// keep track whethere we need to stop the duck from moving
+let shouldStopDuck = false;
+useRequestAnimationFrame(() => {
+  if (!shouldStopDuck) {
+    // update count before each browser repaint
+    const newCount = count >= 100000 ? 0 : count + 1;
+    setCount(newCount);
+  }
+});
+
+obstacleMarkers.forEach((marker) => {
+  const position = marker.pose.position;
+  // make the duck stop when the duck's current position has an obstacle
+  if (position.x === duckPosition.x && position.y === duckPosition.y && position.z === duckPosition.z) {
+    shouldStopDuck = true;
+  }
+});
+s;
+```
+
+Putting it all together, we now have a moving duck that gets stuck on every cube obstacle in its path. Clicking a blocking cube will remove it, allowing the duck to happily continue forward.
```

Comment:
Text is missing explanation of the fact that you can add a cube by clicking a sphere